primefaces-integration-tests/src/test/java/org/primefaces/integrationtests/tree/Tree004Test.java

```diff
@@ -126,10 +131,132 @@ public void testMultipleSelectionWithoutMetakey(Page page) {
+    public void testFilter(Page page) {
+        // Arrange
+        Tree tree = page.tree;
+        Assertions.assertNotNull(tree);
+
+        List<TreeNode> children = tree.getChildren();
+
+        Assertions.assertNotNull(children);
+        Assertions.assertEquals(3, children.size());
+
+        TreeNode first = children.get(0);
+        Assertions.assertEquals("Documents", first.getLabelText());
+
+        Assertions.assertEquals(true, children.get(0).getWebElement().isDisplayed());
+        Assertions.assertEquals(true, children.get(1).getWebElement().isDisplayed());
+        Assertions.assertEquals(true, children.get(2).getWebElement().isDisplayed());
+
+        // Act
+        WebElement filter = tree.findElement(By.cssSelector("input.ui-tree-filter"));
+        ComponentUtils.sendKeys(filter, "Pro");
+        PrimeSelenium.wait(300 + 100); // 300ms are hardcoded as filterDelay in tree.vertical.js
+        PrimeSelenium.waitGui().until(PrimeExpectedConditions.visibleInViewport(tree.findElement(By.id("form:tree:1_1"))));
+
+        // Assert
+        children = tree.getChildren();
+
+        Assertions.assertNotNull(children);
+        Assertions.assertEquals(3, children.size());
+
+        // L1
+        Assertions.assertEquals(false, children.get(0).getWebElement().isDisplayed());
+        Assertions.assertEquals(true, children.get(1).getWebElement().isDisplayed());
+        Assertions.assertEquals(false, children.get(2).getWebElement().isDisplayed());
+        // L2
+        Assertions.assertEquals(false, children.get(1).getChildren().get(0).getWebElement().isDisplayed());
+        Assertions.assertEquals(true, children.get(1).getChildren().get(1).getWebElement().isDisplayed());
+        Assertions.assertEquals(false, children.get(1).getChildren().get(2).getWebElement().isDisplayed());
+
+        // Act
+        filter.clear();
+        ComponentUtils.sendKeys(filter, Keys.BACK_SPACE); // null filter press backspace to trigger the re-filtering
+        PrimeSelenium.wait(300 + 100); // 300ms are hardcoded as filterDelay in tree.vertical.js
+        PrimeSelenium.waitGui().until(PrimeExpectedConditions.visibleInViewport(tree.findElement(By.id("form:tree:0"))));
+
+        // Assert
+        children = tree.getChildren();
+
+        Assertions.assertEquals(true, children.get(0).getWebElement().isDisplayed());
+        Assertions.assertEquals(true, children.get(1).getWebElement().isDisplayed());
+        Assertions.assertEquals(true, children.get(2).getWebElement().isDisplayed());
+
+        assertConfiguration(tree.getWidgetConfiguration());
+    }
+
+    @Test
+    @Order(4)
+    @DisplayName("Tree: Multiple selection + Filter, https://github.com/primefaces/primefaces/issues/3926")
+    public void testFilterWithMultipleSelection(Page page) {
+        // Arrange
+        Tree tree = page.tree;
+        page.selection.select("checkbox");
+        Assertions.assertNotNull(tree);
+
+        List<TreeNode> children = tree.getChildren();
```

Comment:
You have Children as a variable here but but below you repeat yourself with `tree.getChildren()` everywhere making the code harder to read.  Any reason?

Reply:
Nope. I´ll fix this in the evening.

Reply:
Had some more minutes to fix this. Now i switch to famliy-time. :-)